ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketHostAllowlistTest.java

```diff
@@ -200,10 +217,18 @@ private void doHttpRequestAndVerify(
               .result()
               .send(
                   response -> {
-                    assertThat(response.result().statusCode()).isEqualTo(expectedResponse);
-                    testContext.completeNow();
+                    if (response.succeeded()) {
+                      assertThat(response.result().statusCode()).isEqualTo(expectedResponse);
+                      testContext.completeNow();
+                    } else {
+                      testContext.failNow(response.cause());
+                    }
                   });
         });
-    testContext.awaitCompletion(VERTX_AWAIT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    assertThat(testContext.awaitCompletion(VERTX_AWAIT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS))
+        .isTrue();
```

Comment:
@usmansaleem is this what you mean by the test needing fixing? there was no assertion before?

Reply:
@macfarla correct,  from L220 and L232 as well.